### PR TITLE
BXMSPROD-813 githubscm.mergeSourceIntoTarget overwritten for kogito's requirements

### PR DESCRIPTION
**Pull request by Ginxo**
https://issues.redhat.com/browse/BXMSPROD-813

### CODE REVIEW

File: vars/githubscm.groovy [hunk 1]
```diff
@@ -42,36 +42,50 @@ def getRepositoryScm(String repository, String author, String branches) {
...
+        mergeRepo(env.ghprbAuthorRepoGitUrl, sourceBranches, targetAuthor, targetBranches)
+    } else {
+        println "[INFO] ghprbAuthorRepoGitUrl does not exist, CHANGE_FORK used instead '${CHANGE_FORK}'"
+        mergeSourceIntoTarget(repository, env.CHANGE_FORK, sourceBranches, targetAuthor, targetBranches)
+    }
+}
+
+def mergeRepo(String repositoryUrl, String sourceBranches, String targetAuthor, String targetBranches) {
+    println "[INFO] Merging source [${repositoryUrl}:${sourceBranches}] into target [${targetAuthor}/${repository}:${targetBranches}]..."
     checkout(resolveRepository(repository, targetAuthor, targetBranches, false))
```
Comment (MarianMacik) [inline vars/githubscm.groovy:62]: Where is the `repository` coming from?

File: vars/githubscm.groovy [hunk 2]
```diff
@@ -42,36 +42,50 @@ def getRepositoryScm(String repository, String author, String branches) {
     return repositoryScm
 }
 
+def mergeSourceIntoTarget(String repository, String sourceAuthor, String sourceBranches, String targetAuthor, String targetBranches) { 
+    def repositoryUrl = "https://github.com/${sourceAuthor}/${repository}"
+    mergeRepo(repositoryUrl, sourceBranches, targetAuthor, targetBranches)
+}
```
Comment (MarianMacik) [inline vars/githubscm.groovy:48]: This is not used anywhere.

File: vars/githubscm.groovy (hunk 1 above)
Comment (Ginxo) [reply vars/githubscm.groovy]: solved!

File: vars/githubscm.groovy (hunk 2 above)
Comment (Ginxo) [reply vars/githubscm.groovy:48]: is it not used by your pipeline?

Comment (MarianMacik) [reply vars/githubscm.groovy:48]: Nope, I use just githubscm.checkoutIfExists(..)

Comment (MarianMacik) [reply vars/githubscm.groovy:48]: Resolved this now with my commit.

**Pull request merged**